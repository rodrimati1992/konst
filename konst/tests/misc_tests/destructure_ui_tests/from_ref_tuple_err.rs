type Tuple = (u8, u8, u8);

const fn to_bar(foo: &Tuple) -> impl Sized {
    konst::destructure!{(foo, bar, baz) = foo}
    (foo, bar, baz)
}

const fn to_bar_2(foo: &Tuple) -> impl Sized {
    konst::destructure!{(foo, bar, baz): &Tuple = foo}
    (foo, bar, baz)
}


fn main(){}