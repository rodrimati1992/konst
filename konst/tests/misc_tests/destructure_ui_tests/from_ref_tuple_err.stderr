error[E0308]: mismatched types
 --> tests/misc_tests/destructure_ui_tests/from_ref_tuple_err.rs:4:43
  |
4 |     konst::destructure!{(foo, bar, baz) = foo}
  |     --------------------------------------^^^-
  |     |                                     |
  |     |                                     expected `(_, _, _)`, found `&(u8, u8, u8)`
  |     expected due to this
  |
  = note:  expected tuple `(_, _, _)`
          found reference `&(u8, u8, u8)`

error[E0308]: mismatched types
 --> tests/misc_tests/destructure_ui_tests/from_ref_tuple_err.rs:9:5
  |
9 |     konst::destructure!{(foo, bar, baz): &Tuple = foo}
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |     |
  |     expected `(_, _, _)`, found `&(u8, u8, u8)`
  |     arguments to this function are incorrect
  |
  = note:  expected tuple `(_, _, _)`
          found reference `&(u8, u8, u8)`
note: function defined here
 --> src/macros/destructuring.rs
  |
  | pub const fn assert_same_type<T>(this: T, that: T) {
  |              ^^^^^^^^^^^^^^^^
  = note: this error originates in the macro `$crate::__destructure_tuple` which comes from the expansion of the macro `konst::destructure` (in Nightly builds, run with -Z macro-backtrace for more info)
